executor/adapter.go

```diff
@@ -306,7 +306,6 @@ func (a *ExecStmt) Exec(ctx context.Context) (_ sqlexec.RecordSet, err error) {
 	}()
 
 	sctx := a.Ctx
-	ctx = util.SetSessionID(ctx, sctx.GetSessionVars().ConnectionID)
```

Comment:
it seems some other test https://github.com/pingcap/tidb/blob/18cbfaac15f8478902726e1e64971aa96b862462/store/tikv/region_request.go#L413 need this ..

@MyonKeminta  PTAL

Reply:
I have updated the test, you can see the `_test.go` file changes in this PR

Reply:
it seems the previous 
https://github.com/pingcap/tidb/blob/18cbfaac15f8478902726e1e64971aa96b862462/store/tikv/region_request.go#L465
will be used in some test case that out of tidb repo 
https://github.com/pingcap/automated-tests/blob/cc44a56fd59cfdb71eaf37210e0861ba7acdc925/ticases/transaction/asynccommit/linearizability.go#L445

I'm not sure whether it works  😞

Reply:
Seems we may change the session id or abstract a `isInnerExecution` accordingly.

Reply:
Yes that's a problem.  Some failpoints that's used by tests outside the repo checks the session id and works only when session id != 0. These kind of usages also occurs in 2pc.go, prewrite.go, etc.